pkg/controller/jobs/rayjob/rayjob_controller_test.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rayjob
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	rayjobapi "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+	testingrayutil "sigs.k8s.io/kueue/pkg/util/testingjobs/rayjob"
+)
+
+func TestPodSets(t *testing.T) {
```

Comment:
Can we have some simple tests for other functions as well?

Reply:
they are not very complex, also we have integration test for ray

Reply:
I'll try to add some more

Reply:
done